src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -340,7 +340,7 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                 {
                     if (!collectionStopped || ct.IsCancellationRequested)
                     {
-                        ret = ErrorCodes.TracingError;
+                        ret = ReturnCode.TracingError;
```

Comment:
Not 100% sure why as I didn't add this code - this was added in https://github.com/dotnet/diagnostics/pull/1968. 

If I had to make a guess, it'd be because we're terminating a child process before it stopped so we're technically not "cleanly" returning.